crates/prune/src/pruner.rs

```diff
@@ -95,11 +95,16 @@ impl<DB: Database> Pruner<DB> {
         let provider = self.provider_factory.provider_rw()?;
 
         let mut done = true;
-        let mut parts_done = BTreeMap::new();
+        let mut parts = BTreeMap::new();
 
         // TODO(alexey): prune snapshot parts of data (headers, transactions)
         // let highest_snapshots = *self.highest_snapshots_tracker.borrow();
 
+        let mut delete_limit = self.delete_limit *
+            self.last_pruned_block_number
+                .map_or(1, |last_pruned_block_number| tip_block_number - last_pruned_block_number)
+                as usize;
+
```

Comment:
can we just add a doc emphasizing that `last_pruned_block_number` is close to `block_interval` and cannot be too high in the current state?